src/components/share/Default.jsx

```diff
@@ -1,50 +1,80 @@
-import React, { useRef, useState } from 'react';
-import { View } from 'react-native';
+import React, { useRef, useEffect, useState } from 'react';
+import { Keyboard, View } from 'react-native';
+import { useNavigation } from '@react-navigation/native';
 import styled from 'styled-components';
-import { ScrollView, TouchableOpacity } from 'react-native-gesture-handler';
+import { ScrollView, TouchableOpacity, FlatList } from 'react-native-gesture-handler';
 import CommonBottomSheet from '../base/modal/CommonBottomSheet';
 import Button from '../base/Button';
 import Card from '../base/Card';
 import Folder from '../../assets/icons/Folder';
-import Search from '../../assets/icons/IcSearch';
+import IcSearch from '../../assets/icons/IcSearch';
+import { WIDTH } from '../../constants/constants';
 import size from '../../utils/size';
 import CreateFolder from './Create/CreateFolder';
 import SearchFolder from './Search/SearchFolder';
+import SearchInput from '../base/SearchInput';
+import NoSearchResult from './Search/NoSearchResult';
 
-const Default = ({ onPressClose }) => {
+const Default = ({ onClose, onPressNewFolder, onNext }) => {
   const bottomSheetRef = useRef(null);
+  const navigation = useNavigation();
 
-  // const openBottomSheet = () => {
-  //   bottomSheetRef.current?.expand();
-  // };
-
-  // const ref = useRef(null);
-
+  const [showKeyboard, setShowKeyboard] = useState(false);
+  const [doSearch, setDoSearch] = useState(false);
   const [selectedFolder, setSelectedFolder] = useState(null);
   const [openAddFolder, setOpenAddFolder] = useState(false);
   const [openSearchFolder, setOpenSearchFolder] = useState(false);
+  const [searchResults, setSearchResults] = useState([]);
+  const [selected, setSelected] = useState(null);
 
   const folders = [
     { title: ' 폴더명 최대 노출 길이 여...', numberOfLinks: 10 },
     { title: '폴더명 최대 노출 길이 여...', numberOfLinks: 20 },
     { title: '폴더명 최대 노출 길이 여...', numberOfLinks: 30 },
     { title: '폴더명 최대 노출 길이 여...', numberOfLinks: 30 },
+    { title: '폴더명 최대 노출 길이 여...', numberOfLinks: 30 },
+    { title: '폴더명 최대 노출 길이 여...', numberOfLinks: 30 },
+    { title: '폴더명 최대 노출 길이 여...', numberOfLinks: 30 },
+    { title: '폴더명 최대 노출 길이 여...', numberOfLinks: 30 },
+    { title: '폴더명 최대 노출 길이 여...', numberOfLinks: 30 },
+    { title: '폴더명 최대 노출 길이 여...', numberOfLinks: 30 },
   ];
 
-  const handleNext = () => {
-    onPressClose();
-    setOpenAddFolder(true);
+  const handleInputFocus = () => {
+    setShowKeyboard(true);
+    function onKeyboardDidShow(e) {
+      setShowKeyboard(true);
+    }
+
+    const showSubscription = Keyboard.addListener('keyboardDidShow', onKeyboardDidShow);
+    return () => {
+      showSubscription.remove();
+    };
+  };
+
+  const calculateSnapPoints = () => {
```

Comment:
```suggestion
  const calculateSnapPoints = () => {
    let default = 640;
    
    if (showKeyboard) {
      default = 593;
    } else if (doSearch) {
      default = searchResults.length ? 310 : 360;
    }
    
    return [size.height * default]
  }
```

이렇게 작성하는 게 조금 더 깔끔해 보일 듯 합니다.